master/buildbot/worker/base.py

```diff
@@ -461,6 +461,18 @@ def attached(self, conn):
             # reasonable fallback
             self.path_module = namedModule("posixpath")
         log.msg("bot attached")
+
+        #LLVM_LOCAL begin
+        log.msg(f"  host: {workerinfo['host']}")
+        log.msg(f"  admin: {workerinfo['admin']}")
+        log.msg(f"  access_uri: {workerinfo['access_uri']}")
```

Comment:
If for any reason a key is missing, we would get an exception here. How about using `get` with default value "<unspecified>" or something like that?

Reply:
Please look at line 444 above where workerinfo is created. It is guaranteed to have these keys
```
 workerinfo = {
  'admin': conn.info.get('admin'),
  'host': conn.info.get('host'),
  'access_uri': conn.info.get('access_uri'),
  'version': conn.info.get('version')
}
```
Also note yield self.master.data.updates.workerConnected(..., workerinfo=workerinfo) does not change workerinfo.

We can also use `conn.info.get('admin')` directly, but it seems more expensive.

Reply:
Fair enough. `self.master.data.updates.workerConnected` just writes to the database and should not change the `workerinfo` indeed. We can relay on that and crashing is Ok if the assumption gets broken.